#2374 BoundingBox using correct latitude and longitude values

tobrun opened the pull request:
PR for #2374 and #1948
- Fixed BoundingBox using correct latitude and longitude values
- Corrected equals method to be java compliant (BoundingBox and CoordinateSpan)
- Added testcases for BoundingBox
- Replaced intersect to handle non-intersecting boundingboxes
- update jni.cpp to handle new naming of BoundingBox members
- added missing methods for #1948

ljbade commented:
@tobrun I'm confused about the naming of the `BoundingBox` members.

Isn't North/South latitude not longitude? Same with East/West as longitude not latitude

tobrun commented:
The reason why North/South longitude is used, is that these are constant over the whole top/bottom side of the rectangle. If we would take the variable latitude we will have a mismatch using our constructors: (see picture image in #2374)

These 2 constructors should describe the same BoundingBox but are different 
 - BoundingBox( {20,10} , {0,0} ) -> this will call BoundingBox(20,10,0,0)
 - BoundingBox(10,5,0,5)  

Using this PR you will get:
 - BoundingBox( {20,10} , {0,0} ) -> this will call BoundingBox(20,10,0,0)
 - BoundingBox(20,10,0,0)  

Personally I find the longitude/latitude names confusing. I think something like this would make more sense:

- BoundingBox( LatLng north, LatLng east, LatLng south, LatLng west)
- BoundingBox( double longitudeTopSide, double latitudeRightSide, double longitudeBottomSide, double latitudeLeftSide)

ljbade commented:
Still confused.

@bleege @incanus Can you double check this?

tobrun commented:
@ljbade you are right, I made the mistake of thinking that the order of (x,y) in geometry is the same order as LatLng in coordinates. Reverted those changes, this PR boils down to the following:

- added missing methods for #1948
- Intersect will not return new BoundingBox is 2 BoundingBoxed don't intersect 
- Testsuite for BoundingBox
- Java styles equals method for BoundingBox and CoordinateSpan (needed for testcases)

ljbade commented:
@tobrun Can you move this to mapbox repo so it can be merged?

ljbade commented:
Otherwise this is good to merge.
